### DISCUSSION TITLE
Cannot create table using generated code

### DISCUSSION THREAD

**Discussion by Frank1016** [serverpod/serverpod#1350] [Q&A] [answered]
Firstly, I created a User class, and the user table can be created successfully using the auto-generated code. Then I created another class called Address, and it has a field set like this:  "**userId: int, parent=user**",  and the generated code to create address table won't work, showing this error:

![image](https://github.com/serverpod/serverpod/assets/32797520/e15ca173-b64e-419f-9b3b-b1b9a95fb86d)

Is this because the name "User" in conflict with something in Serverpod?

**Comment by Isakdl** [comment, marked as answer]
It is because `user` is a reserved keyword in PostgreSQL (the database Serverpod uses).

We could maybe solve this by escaping the name here, but the quick solution is to use a different name.

**Reply by Frank1016** [reply]
Thanks for the reply! I just searched online for the list of all the reserved keywords in PostgreSQL:
https://www.postgresql.org/docs/current/sql-keywords-appendix.html

**Reply by yahu1031** [reply]
@Isakdl Is there any possibility that we can extend the `USER`?

**Reply by Isakdl** [reply]
@yahu1031 in Postgresql you mean? In postgres `user` refers to the system user of the database (the admin) and not a table or any extendable data. But we should be able to escape the name so that the term `user` can be used in the protocols. 

Basically something like this from the example above
```sql
ALTER TABLE ONLY "address"
  ADD CONSTRAINT address_fk_0
    FOREIGN KEY("userId")
      REFERENCES "user"(id)
        ON DELETE CASCADE;
```
